src/watts/plugin_moose.py

```diff
@@ -205,3 +211,24 @@ def postrun(self, params: Parameters) -> ResultsMOOSE:
         inputs = ['MOOSE.i']
         outputs = [p for p in Path.cwd().iterdir() if p.name not in inputs]
         return ResultsMOOSE(params, time, inputs, outputs)
+
+
+    def _convert_unit(self, params: Parameters):
+        # Function to convert units to SI
+        u.imperial.enable()
+        self.params_copy = copy.deepcopy(params)
```

Comment:
Rather than setting a new attribute, just have this method return the new copy of params. Then in prerun, you'd have:
```
params_copy = self._convert_unit(params)
...
super().prerun(params_copy, ...)